ChatSX/components/Poll/index.tsx

```diff
@@ -0,0 +1,1689 @@
+    setVoteAllowedPerUser(val + 1);
+    hideSelectOptionModal();
+  };
+
+  // this function handles "Cancel" button flow
+  const handleOnCancel = () => {
+    navigation.goBack();
+  };
+
+  // this function hides "User can vote for" modal
+  const hideActionModal = () => {
+    setIsActionAlertModalVisible(false);
+  };
+
+  // this function hides "Number of votes allowed per user" modal
+  const hideSelectOptionModal = () => {
+    setIsOptionAlertModalVisible(false);
+  };
+
+  // this function sets "User can vote for" modal interated value to the state
+  const handleOpenActionModal = () => {
+    let valueArr: any = userCanVoteForArr;
+
+    setUserVoteForOptionsArrValue(valueArr);
+    setIsActionAlertModalVisible(true);
+  };
+
+  const handleShowDateTimePicker = () => {
+    setShow(false);
+    setMode('');
+    setDate('');
+    setTime(new Date());
+  };
+
+  // this function sets "Number of votes allowed per user" modal interated value to the state
+  const handleOpenOptionModal = () => {
+    let quantinyArr: any = [];
+    for (let i = 0; i < optionsArray.length; i++) {
+      quantinyArr = [
+        ...quantinyArr,
+        `${i + 1 > 1 ? `${i + 1} options` : `${i + 1} option`}`,
+      ];
+    }
+    setUserVoteForOptionsArrValue(quantinyArr);
+    setIsOptionAlertModalVisible(true);
+  };
+
+  // this function fetches postPollConversation API
+  async function postPoll() {
+    let expiryTime = !!date ? formatDate(date, time) : null;
+    try {
+      if (question?.trim() === '') {
+        dispatch({
+          type: SHOW_TOAST,
+          body: {isToast: true, msg: QUESTION_WARNING},
+        });
+        return;
+      }
+      if (!expiryTime) {
+        dispatch({
+          type: SHOW_TOAST,
+          body: {isToast: true, msg: EXPIRY_TIME_WARNING},
+        });
+        return;
+      }
+      const tempPollOptionsMap: any = {};
+      let shouldBreak = false;
+      const polls = optionsArray.map((item: any) => {
+        if (tempPollOptionsMap[item?.text] !== undefined) {
+          dispatch({
+            type: SHOW_TOAST,
+            body: {isToast: true, msg: POLLS_OPTIONS_WARNING},
+          });
+          shouldBreak = true;
+        } else {
+          if (item?.text === '') {
+            dispatch({
+              type: SHOW_TOAST,
+              body: {isToast: true, msg: EMPTY_OPTIONS_WARNING},
+            });
+            shouldBreak = true;
+          }
+          tempPollOptionsMap[item?.text] = true;
+          return {
+            text: item?.text,
+          };
+        }
+      });
+      if (shouldBreak) {
+        return;
+      }
+
+      const payload: any = {
+        chatroomId: chatroomID,
+        temporaryId: Date.now().toString(),
+        state: 10,
+        text: question,
+        polls: polls,
+        pollType: liveResultsEnabled ? 1 : 0,
+        multipleSelectState: showAdvancedOption ? userVoteFor : 1,
+        multipleSelectNo: showAdvancedOption ? voteAllowedPerUser : 1,
```

Comment:
Please check request with Android, if user haven't selected any state or number, we should send null here, instead of 1. Please check this with Android or iOS